python/dgl/graphbolt/feature_store.py

```diff
@@ -110,6 +120,29 @@ def size(
         """
         raise NotImplementedError
 
+    def metadata(
```

Comment:
do we need to think more about the `metadata`? In this implementation, metadata is always per triplet of `domain/type_name/feature_name`. do need some metadata that's per `domain`, or even per `feature_store`?

Reply:
Good point, maybe in the future the API can be extended to making domain / type / feature optional.